node/src/actors/chain_manager/mining.rs

```diff
@@ -317,6 +317,7 @@ impl ChainManager {
         // epoch by using a reference-counted atomic counter that can be read and updated safely.
         let current_retrieval_count = Arc::new(AtomicU16::new(0u16));
         let maximum_retrieval_count = self.data_request_max_retrievals_per_epoch;
+        let mut own_utxos_copy = self.chain_state.own_utxos.clone();
```

Comment:
This clone must go away.

This can be achieved by duplicating all the logic of `NodeUtxos` into a `NodeUtxosRef` struct, which is the same but it takes `&OwnUnspentOutputsPool` instead of `&mut OwnUnspentOutputsPool`. I tried that and it increases our line count by 30 lines so I'm not happy with it.